extensions/amp-access-scroll/0.1/scroll-audio.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ScrollComponent} from './scroll-component';
+import {dict} from '../../../src/utils/object';
+import {toggle} from '../../../src/style';
+
+/** Provides iframe for the Scroll Audio Player. */
+export class Audio extends ScrollComponent {
+  /**
+   *  Creates an instance of Audio.
+   *
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} doc
+   */
+  constructor(doc) {
+    super(doc);
+    /** @private {!Audio.State} */
+    this.state_ = {
+      url: '',
+      open: false,
+    };
+  }
+  /**
+   * @param {!JsonObject} action
+   */
+  update(action) {
+    this.state_ = {
+      url: action['url'] !== undefined ? action['url'] : this.state_.url,
+      open: action['open'],
+    };
+
+    this.render_(this.state_);
+  }
+
+  /**
+   * @param {!Audio.State} state
+   * @private
+   */
+  render_(state) {
+    setTimeout(() => {
```

Comment:
Why `setTimeout` here? We may need to switch to `vsync.mutate()` instead.

Reply:
The intention was to make dom updates non-blocking, thanks for pointing out  vsync mutate instead - updated.